pkg/prometheus/agent/statefulset_test.go

```diff
@@ -192,7 +193,7 @@ func newLogger() log.Logger {
 
 func makeStatefulSetFromPrometheus(p monitoringv1alpha1.PrometheusAgent) (*appsv1.StatefulSet, error) {
 	logger := newLogger()
-	cg, err := prompkg.NewConfigGenerator(logger, &p, false)
+	cg, err := prompkg.NewConfigGenerator(context.TODO(), logger, &p, false)
```

Comment:
IIRC we had removed `context.TODO()` in code base a while back, so `context.Background()` would be better here :)